bootstrap/hook.go

```diff
@@ -105,6 +109,14 @@ func newHookScriptWrapper(hookPath string) (*hookScriptWrapper, error) {
 			"SET " + hookWorkingDirEnv + "=%CD%\n" +
 			"SET > \"" + h.afterEnvFile.Name() + "\"\n" +
 			"EXIT %" + hookExitStatusEnv + "%"
+	} else if runtime.GOOS == "windows" && isPwshHook {
+		script = "$ErrorActionPreference = \"STOP\"\n" +
+			"Get-ChildItem Env: | Foreach-Object {\"$($_.Name)=$($_.Value)\"} | Set-Content \"" + h.beforeEnvFile.Name() + "\"\n" +
+			absolutePathToHook + "\n" +
+			"if ($LASTEXITCODE -eq $null) {$Env:" + hookExitStatusEnv + " = 0} else {$Env:" + hookExitStatusEnv + " = $LASTEXITCODE}\n" +
+			"$Env:" + hookWorkingDirEnv + " = $PWD | Select-Object -ExpandProperty Path\n" +
+			"Get-ChildItem Env: | Foreach-Object {\"$($_.Name)=$($_.Value)\"} | Set-Content \"" + h.afterEnvFile.Name() + "\"\n" +
+			"exit $Env:" + hookExitStatusEnv
```

Comment:
Would Go's backtick-strings work here to (a) reduce escaping and (b) help distinguish between outer-Go-strings and inner-PS-strings? I think it would make it quite a bit more readable. Like this:

```diff
-		script = "$ErrorActionPreference = \"STOP\"\n" +
-			"Get-ChildItem Env: | Foreach-Object {\"$($_.Name)=$($_.Value)\"} | Set-Content \"" + h.beforeEnvFile.Name() + "\"\n" +
-			absolutePathToHook + "\n" +
-			"if ($LASTEXITCODE -eq $null) {$Env:" + hookExitStatusEnv + " = 0} else {$Env:" + hookExitStatusEnv + " = $LASTEXITCODE}\n" +
-			"$Env:" + hookWorkingDirEnv + " = $PWD | Select-Object -ExpandProperty Path\n" +
-			"Get-ChildItem Env: | Foreach-Object {\"$($_.Name)=$($_.Value)\"} | Set-Content \"" + h.afterEnvFile.Name() + "\"\n" +
-			"exit $Env:" + hookExitStatusEnv
+		script = `$ErrorActionPreference = "STOP"\n` +
+			`Get-ChildItem Env: | Foreach-Object {"$($_.Name)=$($_.Value)"} | Set-Content "` + h.beforeEnvFile.Name() + `"\n` +
+			absolutePathToHook + `\n` +
+			`if ($LASTEXITCODE -eq $null) {$Env:` + hookExitStatusEnv + ` = 0} else {$Env:` + hookExitStatusEnv + ` = $LASTEXITCODE}\n` +
+			`$Env:` + hookWorkingDirEnv + ` = $PWD | Select-Object -ExpandProperty Path\n` +
+			`Get-ChildItem Env: | Foreach-Object {"$($_.Name)=$($_.Value)"} | Set-Content "` + h.afterEnvFile.Name() + `"\n` +
+			`exit $Env:` + hookExitStatusEnv
```